File: C++ Code/Basics/Array/1 D/Insertion sort.cpp
#include<iostream>
void ascarr(int arr[], int n){
    for(int i=1;i<n;i++){
        int temp=arr[i],j=i-1;
        while(j>=0){
            if(arr[j]>temp){
                arr[j+1]=arr[j];
                j--;
            }
            else
                break;
        }
        arr[j+1]=temp;
    }
}
void desarr(int arr[], int n){
    for(int i=1;i<n;i++){
        int temp=arr[i],j=i-1;
        while(j>=0){
            if(arr[j]<temp){
                arr[j+1]=arr[j];
                j--;
            }
            else
                break;
        }
        arr[j+1]=temp;
    }
}
int main(){
    int n,x;
    std::cout<<"Enter the size of array : ";
    std::cin>>n;
    int arr[n];
    std::cout<<"Enter the value of elements : "<<std::endl;
    for(int i=0;i<n;i++)
        std::cin>>arr[i];
    std::cout<<"Press 1 or 2 (1 for ascending and 2 for descending order of sorted array) : ";
    std::cin>>x;
    switch(x){
        case 1:
            ascarr(arr,n);
            std::cout<<"Incresing order of sorted array : "<<std::endl;
            break;
        case 2:
            desarr(arr,n);
            std::cout<<"Decresing order of sorted array : "<<std::endl;
            break;
        default :
            std::cout<<"Invalid option.";
    }
    for(int i=0;i<n;i++)
        std::cout<<arr[i]<<"  ";
    return 0;
}

File: Java Code/Basics/Array/1 D/InsertionSort.java
import java.util.Scanner;

public class InsertionSort{
  public static void main(String[] args){
    Scanner sc=new Scanner(System.in);
    System.out.print("Enter the size of array : ");
    int n=sc.nextInt();
    int[] arr=new int[n];
    System.out.println("Enter all the elements :");
    for(int i=0;i<n;i++)
      arr[i]=sc.nextInt();
    sort(arr);
    System.out.println("Sorted array :");
    for(int i=0;i<n;i++)
      System.out.print(arr[i]+" ");
    System.out.println(" ");
  }
  public static void sort(int[] arr){
    for(int i=0;i<arr.length;i++){
      int j=i;
      while(j>0 && arr[j]<arr[j-1]){
        if(arr[j-1]>arr[j]){
          int temp=arr[j-1];
          arr[j-1]=arr[j];
          arr[j]=temp;
          j--;
        }
      }
    }
  }
}